components/announcement_bar/configuration_bar/configuration_bar.tsx

```diff
@@ -247,18 +250,22 @@ const ConfigurationAnnouncementBar = (props: Props) => {
 
             let announcementBarType = AnnouncementBarTypes.ANNOUNCEMENT;
 
+            const {w} = getViewportSize();
             if (daysUntilLicenseExpires < 1) {
+                const viewportBasedMessage = w < Constants.MOBILE_SCREEN_WIDTH ? formatMessage({
+                    id: 'announcement_bar.error.trial_license_expiring_last_day.short',
+                    defaultMessage: 'This is the last day of your free trial.'},
+                ) : formatMessage({
+                    id: 'announcement_bar.error.trial_license_expiring_last_day',
+                    defaultMessage: 'This is the last day of your free trial. Purchase a license now to continue using Mattermost Professional and Enterprise features.',
+                });
                 message = (
                     <>
                         <img
                             className='advisor-icon'
                             src={warningIcon}
                         />
-                        <FormattedMessage
-                            id='announcement_bar.error.trial_license_expiring_last_day'
-                            tagName='strong'
-                            defaultMessage={'This is the last day of your free trial. Purchase a license now to continue using Mattermost Professional and Enterprise features.'}
-                        />
+                        <strong>{viewportBasedMessage}</strong>
```

Comment:
nit: Solving the issue by reducing the message text could not work in other languages. Just make sure UX is ok with it.

Reply:
Its Ok. confirmed from @annelieseklein